esp_hosted_ng/host/include/esp_if.h

```diff
@@ -29,7 +29,18 @@ struct esp_if_ops {
 	int (*deinit)(struct esp_adapter *adapter);
 };
 
-int esp_init_interface_layer(struct esp_adapter *adapter);
-void esp_deinit_interface_layer(void);
+struct esp_if_params {
+	uint8_t speed;
+	uint32_t handshake_pin;
+	uint32_t chip_select;
+	uint32_t data_ready_pin;
+};
+
+int esp_init_interface_layer_spi(struct esp_adapter *adapter, const struct esp_if_params *params);
```

Comment:
no need to make spi and sdio together. it will always be only one at any point of time.
This is the reason declaration are in common places and definitions are in transport specific file.